x-pack/plugins/index_management/public/application/components/component_templates/component_template_wizard/component_template_form/component_template_form.tsx

```diff
@@ -137,37 +134,63 @@ export const ComponentTemplateForm = ({
     </>
   ) : null;
 
-  const buildComponentTemplateObject = (initialTemplate: ComponentTemplateDeserialized) => (
-    wizardData: WizardContent
-  ): ComponentTemplateDeserialized => {
-    const componentTemplate = {
-      ...initialTemplate,
-      name: wizardData.logistics.name,
-      version: wizardData.logistics.version,
-      _meta: wizardData.logistics._meta,
-      template: {
-        settings: wizardData.settings,
-        mappings: wizardData.mappings,
-        aliases: wizardData.aliases,
-      },
-    };
-    return componentTemplate;
+  /**
+   * If no mappings, settings or aliases are defined, it is better to not send an empty
+   * object for those values.
+   * @param componentTemplate The component template object to clean up
+   */
+  const cleanupComponentTemplateObject = (componentTemplate: ComponentTemplateDeserialized) => {
+    const outputTemplate = { ...componentTemplate };
+
+    if (outputTemplate.template.settings === undefined) {
```

Comment:
Could there ever be a case where "settings", "mappings" or "aliases" is an empty object at this point? If so we should probably also delete that from the output template.

Reply:
👍 It is already being stripped out here: https://github.com/elastic/kibana/blob/master/x-pack/plugins/index_management/public/application/components/shared/components/wizard_steps/use_json_step.ts#L44